spec/services/participant_units_builder_spec.rb

```diff
@@ -52,6 +52,60 @@
         expect(Rollbar).not_to have_received(:warning)
       end
 
+      describe 'prioritizes the right roles' do
+        shared_examples 'choosing the right role' do |given_roles, expected_role|
+          subject(:role) { unit_participants.first.role }
+
+          let(:participants_data) do
+            # Attention: this is a very simplified representation of the actual export with the fields needed
+            [
+              {
+                roles: roles.map { |r| { type: r } },
+                links: { person: '3186' }
+              }
+            ].as_json
+          end
+
+          let(:expected_role) { expected_role }
+
+          context "when given #{given_roles.join(',')}" do
+            let(:roles) { given_roles }
+
+            it { is_expected.to eq expected_role }
+          end
+
+          context "when comes #{given_roles.reverse.join(',')}" do
+            let(:roles) { given_roles.reverse }
+
+            it { is_expected.to eq expected_role }
+          end
+        end
+
+        it_behaves_like 'choosing the right role',
+                        ['Event::Camp::Role::AssistantLeader', 'Event::Camp::Role::Participant'],
+                        'assistant_leader'
+
+        it_behaves_like 'choosing the right role',
+                        ['Event::Camp::Role::AssistantLeader', 'Event::Camp::Role::Helper'],
+                        'assistant_leader'
+
+        it_behaves_like 'choosing the right role',
+                        ['Event::Camp::Role::LeaderMountainSecurity', 'Event::Camp::Role::Participant'],
+                        'leader_mountain_security'
```

Comment:
i would nevertheless import this one and just put it differently on the badge. but your call 👍